kv/kvserver: TestReplicateQueueUpAndDownReplicateNonVoters failed

kv/kvserver.TestReplicateQueueUpAndDownReplicateNonVoters [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4141073&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4141073&tab=artifacts#/) on master @ [365b4da8bd02c06ee59d2130a56dec74ffc9ce21](https://github.com/cockroachdb/cockroach/commits/365b4da8bd02c06ee59d2130a56dec74ffc9ce21):


```
=== RUN   TestReplicateQueueUpAndDownReplicateNonVoters
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateQueueUpAndDownReplicateNonVoters2210080764
    test_log_scope.go:80: use -show-logs to present logs inline
    replicate_queue_test.go:359: condition failed to evaluate within 45s: expected upreplication to 2 non-voters; found 0
        goroutine 96551138 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x538caf8, 0xc01a2b4d00}, 0x0, 0xc016d840c0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x538caf8, 0xc01a2b4d00}, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestReplicateQueueUpAndDownReplicateNonVoters(0xc01a2b4d00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replicate_queue_test.go:359 +0x4c5
        testing.tRunner(0xc01a2b4d00, 0x43da390)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestReplicateQueueUpAndDownReplicateNonVoters2210080764
--- FAIL: TestReplicateQueueUpAndDownReplicateNonVoters (46.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=2
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueUpAndDownReplicateNonVoters.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -331,7 +331,16 @@ func TestReplicateQueueUpAndDownReplicateNonVoters(t *testing.T) {
 
 	ctx := context.Background()
 	tc := testcluster.StartTestCluster(t, 1,
-		base.TestClusterArgs{ReplicationMode: base.ReplicationAuto},
+		base.TestClusterArgs{
+			ReplicationMode: base.ReplicationAuto,
+			ServerArgs: base.TestServerArgs{
+				Knobs: base.TestingKnobs{
+					SpanConfig: &spanconfig.TestingKnobs{
+						ConfigureScratchRange: true,
+					},
+				},
+			},
+		},
 	)
 	defer tc.Stopper().Stop(context.Background())
 
@@ -833,6 +842,11 @@ func TestReplicateQueueShouldQueueNonVoter(t *testing.T) {
 					{
 						Key: "rack", Value: strconv.Itoa(i),
 					},
+				},
+			},
+			Knobs: base.TestingKnobs{
+				SpanConfig: &spanconfig.TestingKnobs{
+					ConfigureScratchRange: true,
 				},
 			},
 		}

